homeassistant/components/nmbs/sensor.py

```diff
@@ -124,7 +135,7 @@ def name(self):
     @property
     def unique_id(self):
         """Return a unique ID."""
-        unique_id = f"{self._station}_{self._station_from}_{self._station_to}"
```

Comment:
Open question: How to migrate entities that are not associated to a config flow? 
This is the "liveboard" mode that tracks departing trains in a specific station, and therefor no need to include a from and to station. Earlier this was part because it was configured together with the "connection" mode.

Reply:
Why not replicating the "previous behavior" and just set up both sensors every time?
Meaning keep from and to required and set up the live board directly (possibly disabled by default)?

Reply:
In the previous case, the second "liveboard" sensor is only setup when you fill in the parameter station_live. If it's not filled in, it cannot be setup as well. 

I've added the from and to station back to the "liveboard" sensor now, that way there's no migration needed right now. Later on, the unique id can be migrated given that they will be associated to a config flow entry.

Reply:
I think you should always setup the liveboard.
If the user had it before in the yaml then the entity is enabled, if the user didn't then create it disabled.

Then we can skip the menu in the first step and secondly the user doesn't have to create two config entries for the same station. It would also make the whole process simpler for the user.

Reply:
I'm not sure if you are following the concept completely. The liveboard option is the yaml is optional. The entity cannot be setup if the user doesn't provide the station. A liveboard is s0ecific to a station. 

A connection from a to b is being setup, that doesn't mean that the additional and optional liveboard is either a or b. Which make it harder to detect if entity or configuration flow are already configured and to ensure that entities across the integration are truly unique. 

Config flow 1: connection a to b, with liveboard c 
Config flow 2: connection a to b, with liveboard a
Config flow 3: connection a to d, with liveboard a

This specific case is already complex, but the main reason I pulled them apart. Liveboard a would be provided by config flow 2 and 3, I should prevent that from happening.  

I'm guessing that not many users do use the liveboard to be honest. But there's ni way to find out.

Reply:
We don't need to (or even should in this case) do same for imported yaml vs. if someone sets up the integration from UI.

May I suggest the following:
- Import from yaml: Setup the liveboard automatically as imported from yaml (skip obviously if not part of the yaml config)
- New from UI: Automatically create liveboard entities for both stations but create them as disabled entities.

That way we can slim the config flow as mentioned before to make it simpler for the user and for those who configured from yaml they still get their entities as before.

Reply:
How can I make sure that the entities with same unique id are not provided by multiple config flows? For example, if you set up connection a=>b and b=>c as 2 config flows.

Reply:
Either check if it already exist and therefore don't set it up twice (preferred solution I think) or prefix the unique id with `entry_id` (but then you need to migrate the entities to the new unique id as part of the import)